framework/doc/content/finite_volumes/fv_design.md

```diff
@@ -0,0 +1,173 @@
+they still lack test/weight functions.  FV Dirichlet BCs, however must be
+implemented completedly differently than in FE and strongly motivate the
+creation of a separate FV BC system.
+
+Dirichlet boundary conditions (BCs) in an FV method cannot be created by
+directly setting degrees of freedom like in an FE method because the FV
+degrees of freedom do not exist on the mesh boundary.  There are various
+approaches for dealing with this in FV.  A ghost-element approach was selected
+due to its popularity and robustness.  In this approach, Dirichlet BCs are
+implemented as a weak BC.  To do this, the normal flux kernel terms are
+applied at the mesh boundary faces.  Since flux kernels are calculated using
+information from cells on both sides of the face, we use the desired Dirichlet
+BC value to extrapolate a "ghost" cell value for the side of the face that has
+no actuall mesh cell.  Other necessary cell properties are also
+reflected/mirrored from the existing cell.  A design was chosen that allows
+handling ghost-element creation and use by existing flux kernels automatically
+for enforcement of Dirichlet BCs.  This procedure results in the Dirichlet BC
+objects not actually being respondible for calculating residual contributions.
+They instead inform the ghost-element initialization while the normal flux BCs
+are used to calculate boundary residual contributions.  This and other
+differences motivated the creation of a separate FV BC system.
+
+Many objects in MOOSE get information about the current mesh location by
+directly accessing their own and inherited member variables.  This becomes
+somewhat tricky to handle for FV because of the nature of ghost elements.
+Traditionally, if an object needed the cell volume, it would access an element
+pointer and use libMesh APIs.  This doesn't work for elements that don't
+exist, yet we still need to provide this information for calculating Dirichlet
+BC residual contributions.  We need to be able to provide information that
+doesn't exist and we need to make sure code doesn't try to access irrelevant
+or wrong information directly from the assembly, FEProblem, or other classes.
+For this reason, among others, for FV objects the convention has been
+established for objects to retrieve needed information from a face info object
+that is passed around rather than retrieving binding references to the usual
+mesh-related data.  If everyone gets needed information from this one place -
+it is easy to monitor when code may be doing the wrong thing.  It also becomes
+a simple matter to provide volumes for non-existing cells and add features
+that require intercepting and modifying any face information objects need.
+
+## Reconstruction
+
+TODO: implement reconstruction and discuss design decisions here.
+
+## Known Limitations/Issues
+
+* FE <---> FV variable-coupling does not work.  Particularly, if FE
+  variables try to couple to an FV variable, they will segfault when trying to
```

Comment:
This is partially addressed.  Except in more sophisticated cases where variable values are routed through material properties instead of coupled to directly from FE objects.  Fixing that final case is very difficult.